### ISSUE TITLE
Using client after dropped connection leads to freezing of the whole NodeJs Process

### ISSUE THREAD

**Issue by lokoArt** [grpc/grpc-node#1490]
### Problem description
Please correct me if I'm wrong, this grpc client should automatically reconnect if the connection becomes invalid.
I see this behavior in most cases, however there is one error which makes the client unusable and the whole NodeJs app gets frozen. It' s "Connection dropped". Other errors such as "Channel has been shut down", "Received RST_STREAM with code 2" don't lead to such behavior. 

### Reproduction steps
I create a client at a start of my app, then I occasionally use this grpc client. It works for a few hours but then all of the sudden my app gets frozen. And the last log is always this:

`Connection dropped","metadata":{"internalRepr":{},"options":{}},"level":"warn","message":"Failed to execute grpc callConnection dropped","stack":"Error: Connection dropped\n    at Object.callErrorFromStatus (/usr/src/app/node_modules/@grpc/grpc-js/src/call.ts:81:24)\n    at Object.onReceiveStatus (/usr/src/app/node_modules/@grpc/grpc-js/src/client.ts:334:36)\n    at Object.onReceiveStatus (/usr/src/app/node_modules/@grpc/grpc-js/src/client-interceptors.ts:434:34)\n    at Object.onReceiveStatus (/usr/src/app/node_modules/@grpc/grpc-js/src/client-interceptors.ts:397:48)\n    at Http2CallStream.outputStatus (/usr/src/app/node_modules/@grpc/grpc-js/src/call-stream.ts:230:22)\n    at Http2CallStream.maybeOutputStatus (/usr/src/app/node_modules/@grpc/grpc-js/src/call-stream.ts:280:14)\n    at Http2CallStream.endCall (/usr/src/app/node_modules/@grpc/grpc-js/src/call-stream.ts:263:12)\n    at Http2CallStream.disconnectListener (/usr/src/app/node_modules/@grpc/grpc-js/src/call-stream.ts:215:12)\n    at Socket.session.socket.once (/usr/src/app/node_modules/@grpc/grpc-js/src/subchannel.ts:518:13)\n    at Object.onceWrapper (events.js:286:20)\n    at Socket.emit (events.js:203:15)\n    at Socket.EventEmitter.emit (domain.js:448:20)\n    at TCP._handle.close (net.js:607:12)`

### Temporarily fix
Now if I get an exception during executing a grpc call I just close the current client and create a new one. But I wonder if it's the right approach. 

### Environment
 - Debian GNU/Linux 9 (stretch)
 - Node 10
 - Node installation method: apt
 - "@grpc/grpc-js": "^1.1.1",

### Additional context
Add any other context about the problem here. If possible, attach full logs. Do not try to omit anything for brevity, but instead include absolutely everything. Please try and set your operating system's locale to English, so that logs contain error messages in the English language.

**Comment by murgatroid99** [comment]
Can you replicate this failure with the environment variables `GRPC_TRACE=all` and `GRPC_VERBOSITY=DEBUG` and share the log, or at least the last thousand lines or so?

Also, can you elaborate on what you mean when you say that it makes the client "unusable"? What API are you calling, what behavior do you expect, and what behavior do you see instead?

**Comment by lokoArt** [comment]
> Can you replicate this failure with the environment variables `GRPC_TRACE=all` and `GRPC_VERBOSITY=DEBUG` and share the log, or at least the last thousand lines or so?

Okay, I will write a test in the next few days and try to isolate the issue and provide the code


> Also, can you elaborate on what you mean when you say that it makes the client "unusable"? What API are you calling, what behavior do you expect, and what behavior do you see instead?

I mean when all of the sudden we cannot execute a certain gRpc endpoint due to connectivity issues.

**Comment by murgatroid99** [comment]
> I mean when all of the sudden we cannot execute a certain gRpc endpoint due to connectivity issues.

What API are you calling, what behavior do you expect, and what behavior do you see instead?

**Comment by pedramphp** [comment]
I am also facing similar issue.
when a client stream connection fails after deadline exceeds, the session because unusable and all subsequent calls fail regardless if its unary or streaming call.

Ideally, if there is anything wrong with the session, it has to get restarted before we invoke a request on it.
Please prioritize this issue.

I made the following change and it started working.

https://github.com/grpc/grpc-node/blob/master/packages/grpc-js/src/subchannel.ts#L705
```
this.startConnectingInternal();
try { http2Stream = this.session.request(headers); }
```

**Comment by murgatroid99** [comment]
What version of grpc-js are you using? After one call fails, in what way do the subsequent calls fail? What specific error are you seeing?

**Comment by ganemone** [comment]
After the first call fails with deadline exceeds, all following requests will also fail with deadline exceeds. It seems like this is because the subsequent requests never are actually sent.

**Comment by murgatroid99** [comment]
OK, can you run your client with the environment variables `GRPC_TRACE=all` and `GRPC_VERBOSITY=DEBUG`, and share the output of a run that fails that way?

**Comment by murgatroid99** [comment]
Also, what version of grpc-js are you using?

**Comment by pedramphp** [comment]
I am using grpc-js version `10.21.0` and Node version `10.21.0`
After the Client streaming RPC call timesout, subsequent requests Unary RPC or Client streaming RPC all start timing out because the main http2 session is on a broken state.
I checked the logs and they all looked normal, but the requests never gets sent.

I will run your client with the environment variables GRPC_TRACE=all and GRPC_VERBOSITY=DEBUG, and share the output.

**Comment by murgatroid99** [comment]
The grpc-js library does not have a version 10.21.0. What version of grpc-js are you using?

**Comment by pedramphp** [comment]
Fixed typo `1.2.8`

**Comment by takaaki7** [comment]
I'm facing same issue.
When calling `runPartitionedUpdate()` of Spanner client, method never return or takes too long time(about 1hour or higher). 
Using `"@google-cloud/spanner": "5.8.0"`, `"@grpc/grpc-js": "1.3.2"`
This problem occurs frequently, and started to happen since 5/11 suddenly. 

I've switched to use Java, the problem is resolved.

As far as I see logs with debug options(`GRC_TRACE=all`, `GRPC_VERBOSITY=DEBUG`), when this problem is occurring, server sent GOAWAY.
`grpc-js` doesn't handle GOAWAY properly?


<details>
<summary>Success case log</summary>
```
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.975Z | call_stream | [2] close http2 stream with code 0
2021-06-21T03:52:19.341Z,0.log,"2021-06-21T03:52:18.975Z | call_stream | [2] ended with status: code=0 details="""""
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.975Z | call_stream | [2] received status code 0 from server
2021-06-21T03:52:19.341Z,0.log,grpc-server-stats-bin: AABiqYcUAAAAAA
2021-06-21T03:52:19.341Z,0.log,grpc-status: 0
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.975Z | call_stream | [2] Received server trailers:
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.973Z | call_stream | [2] pushing to reader message of length 5
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.973Z | call_stream | [2] filterReceivedMessage of length 10
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.973Z | call_stream | [2] parsed message of length 10
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.972Z | call_stream | [2] receive HTTP/2 data frame of length 10
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.636Z | call_stream | [2] pushing to reader message of length 53
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.636Z | call_stream | [2] filterReceivedMessage of length 58
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.635Z | call_stream | [2] parsed message of length 58
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.635Z | call_stream | [2] receive HTTP/2 data frame of length 58
2021-06-21T03:52:19.341Z,0.log,server-timing: gfet4t7; dur=8
2021-06-21T03:52:19.341Z,0.log,"date: Mon, 21 Jun 2021 03:52:18 GMT"
2021-06-21T03:52:19.341Z,0.log,"accept-encoding: identity,gzip"
2021-06-21T03:52:19.341Z,0.log,"grpc-accept-encoding: identity,deflate,gzip"
2021-06-21T03:52:19.341Z,0.log,content-type: application/grpc
2021-06-21T03:52:19.341Z,0.log,content-disposition: attachment
2021-06-21T03:52:19.341Z,0.log,:status: 200
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.635Z | call_stream | [2] Received server headers:
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.625Z | subchannel | 2404:6800:4008:c03::5f:443 TRANSIENT_FAILURE -> IDLE
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.625Z | subchannel | 2404:6800:4008:c03::5f:443 CONNECTING -> TRANSIENT_FAILURE
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.625Z | subchannel | 2404:6800:4008:c03::5f:443 connection closed with error connect ENETUNREACH xxx - Local (:::0)
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.624Z | subchannel | 2404:6800:4008:c03::5f:443 TRANSIENT_FAILURE -> CONNECTING
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.624Z | call_stream | [2] calling end() on HTTP/2 stream
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.623Z | call_stream | [2] sending data chunk of length 403 (deferred)
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.623Z | subchannel_refcount | xxx callRefcount 0 -> 1
2021-06-21T03:52:19.341Z,0.log,2021-06-21T03:52:18.623Z | call_stream | [2] attachHttp2Stream from subchannel xxx
2021-06-21T03:52:19.340Z,0.log,te: trailers
2021-06-21T03:52:19.340Z,0.log,:path: /google.spanner.v1.Spanner/ExecuteStreamingSql
2021-06-21T03:52:19.340Z,0.log,:method: POST
2021-06-21T03:52:19.340Z,0.log,content-type: application/grpc
2021-06-21T03:52:19.340Z,0.log,user-agent: grpc-node-js/1.3.2
2021-06-21T03:52:19.340Z,0.log,:authority: spanner.googleapis.com:443
2021-06-21T03:52:19.340Z,0.log,accept-encoding: identity
2021-06-21T03:52:19.340Z,0.log,"grpc-accept-encoding: identity,deflate,gzip"
2021-06-21T03:52:19.340Z,0.log,grpc-timeout: 17999806m
2021-06-21T03:52:19.340Z,0.log,authorization: Bearer xxx
2021-06-21T03:52:19.340Z,0.log,x-goog-request-params: xxx
2021-06-21T03:52:19.340Z,0.log,google-cloud-resource-prefix: xxx
2021-06-21T03:52:19.340Z,0.log,x-goog-api-client: gax/2.14.1 gapic/5.8.0 gl-node/14.15.4 grpc/ gccl/5.8.0
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.623Z | call_stream | Starting stream on subchannel xxx with headers
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.623Z | call_stream | [2] deferring writing data chunk of length 403
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.622Z | call_stream | [2] end() called
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.622Z | call_stream | [2] write() called with message of length 398
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.520Z | channel | Pick result: COMPLETE subchannel: xxx status: undefined undefined
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.520Z | call_stream | [2] Sending metadata
2021-06-21T03:52:19.340Z,0.log,"2021-06-21T03:52:18.520Z | channel | dns:spanner.googleapis.com:443 createCall [2] method=""/google.spanner.v1.Spanner/ExecuteStreamingSql"", deadline=Mon Jun 21 2021 08:52:18 GMT+0000 (Coordinated Universal Time)"
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.343Z | call_stream | [1] HTTP/2 stream closed with code 0
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.343Z | subchannel_refcount | xxx callRefcount 1 -> 0
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.343Z | call_stream | [1] close http2 stream with code 0
2021-06-21T03:52:19.340Z,0.log,"2021-06-21T03:52:18.343Z | call_stream | [1] ended with status: code=0 details="""""
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.342Z | call_stream | [1] received status code 0 from server
2021-06-21T03:52:19.340Z,0.log,grpc-server-stats-bin: xxx
2021-06-21T03:52:19.340Z,0.log,grpc-status: 0
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.342Z | call_stream | [1] Received server trailers:
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.341Z | call_stream | [1] pushing to reader message of length 40
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.341Z | call_stream | [1] filterReceivedMessage of length 45
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.340Z | call_stream | [1] parsed message of length 45
2021-06-21T03:52:19.340Z,0.log,2021-06-21T03:52:18.340Z | call_stream | [1] receive HTTP/2 data frame of length 45
2021-06-21T03:52:19.340Z,0.log,server-timing: gfet4t7; dur=8
2021-06-21T03:52:19.340Z,0.log,"date: Mon, 21 Jun 2021 03:52:18 GMT"
2021-06-21T03:52:19.340Z,0.log,"accept-encoding: identity,gzip"
2021-06-21T03:52:19.340Z,0.log,"grpc-accept-encoding: identity,deflate,gzip"
2021-06-21T03:52:19.340Z,0.log,content-type: application/grpc
2021-06-21T03:52:19.338Z,0.log,content-disposition: attachment
2021-06-21T03:52:19.338Z,0.log,:status: 200
2021-06-21T03:52:19.338Z,0.log,2021-06-21T03:52:18.340Z | call_stream | [1] Received server headers:
2021-06-21T03:52:18.981Z,0.log,succeeded to process pdml
2021-06-21T03:52:18.338Z,0.log,2021-06-21T03:52:18.330Z | call_stream | [1] calling end() on HTTP/2 stream
2021-06-21T03:52:18.338Z,0.log,2021-06-21T03:52:18.330Z | call_stream | [1] sending data chunk of length 165 (deferred)
2021-06-21T03:52:18.338Z,0.log,2021-06-21T03:52:18.330Z | subchannel_refcount | xxx callRefcount 0 -> 1
2021-06-21T03:52:18.338Z,0.log,2021-06-21T03:52:18.329Z | call_stream | [1] attachHttp2Stream from subchannel xxx
2021-06-21T03:52:18.338Z,0.log,te: trailers
2021-06-21T03:52:18.338Z,0.log,:path: /google.spanner.v1.Spanner/BeginTransaction
2021-06-21T03:52:18.338Z,0.log,:method: POST
2021-06-21T03:52:18.338Z,0.log,content-type: application/grpc
2021-06-21T03:52:18.338Z,0.log,user-agent: grpc-node-js/1.3.2
2021-06-21T03:52:18.338Z,0.log,:authority: spanner.googleapis.com:443
2021-06-21T03:52:18.338Z,0.log,accept-encoding: identity
2021-06-21T03:52:18.338Z,0.log,"grpc-accept-encoding: identity,deflate,gzip"
2021-06-21T03:52:18.338Z,0.log,grpc-timeout: 59996m
2021-06-21T03:52:18.338Z,0.log,authorization: xxx
2021-06-21T03:52:18.338Z,0.log,x-goog-request-params: xxx
2021-06-21T03:52:18.338Z,0.log,google-cloud-resource-prefix: xxx
2021-06-21T03:52:18.338Z,0.log,x-goog-api-client: gax/2.14.1 gapic/5.8.0 gl-node/14.15.4 grpc/ gccl/5.8.0
2021-06-21T03:52:18.338Z,0.log,2021-06-21T03:52:18.329Z | call_stream | Starting stream on subchannel 64.233.189.95:443 with headers
2021-06-21T03:52:18.338Z,0.log,2021-06-21T03:52:18.329Z | call_stream | [1] deferring writing data chunk of length 165
2021-06-21T03:52:18.338Z,0.log,2021-06-21T03:52:18.329Z | call_stream | [1] end() called
2021-06-21T03:52:18.338Z,0.log,2021-06-21T03:52:18.329Z | call_stream | [1] write() called with message of length 160
```
</details>

<details> 
<summary>Failure case log(method never returned)</summary>
```
2021-06-21T07:34:29.635Z,2021-06-21T07:34:29.248Z | subchannel_refcount | xxx refcount 1 -> 0
2021-06-21T07:34:29.635Z,2021-06-21T07:34:29.248Z | subchannel_refcount | xxx refcount 1 -> 0
2021-06-21T07:34:29.635Z,2021-06-21T07:34:29.247Z | subchannel_refcount | xxx refcount 1 -> 0
2021-06-21T07:34:25.635Z,2021-06-21T07:34:25.237Z | connectivity_state | dns:pubsub.googleapis.com:443 READY -> IDLE
2021-06-21T07:34:25.635Z,2021-06-21T07:34:25.237Z | resolving_load_balancer | dns:pubsub.googleapis.com:443 READY -> IDLE
2021-06-21T07:34:25.635Z,2021-06-21T07:34:25.237Z | pick_first | READY -> IDLE
2021-06-21T07:34:25.635Z,2021-06-21T07:34:25.237Z | subchannel_refcount | xxx refcount 2 -> 1
2021-06-21T07:34:25.635Z,2021-06-21T07:34:25.237Z | subchannel | xxx READY -> IDLE
2021-06-21T07:34:25.635Z,2021-06-21T07:34:25.237Z | subchannel | xxx connection closed by GOAWAY with code 0
2021-06-21T07:34:23.634Z,2021-06-21T07:34:23.203Z | connectivity_state | dns:pubsub.googleapis.com:443 READY -> IDLE
2021-06-21T07:34:23.634Z,2021-06-21T07:34:23.203Z | resolving_load_balancer | dns:pubsub.googleapis.com:443 READY -> IDLE
2021-06-21T07:34:23.634Z,2021-06-21T07:34:23.203Z | pick_first | READY -> IDLE
2021-06-21T07:34:23.634Z,2021-06-21T07:34:23.203Z | subchannel_refcount | xxx refcount 2 -> 1
2021-06-21T07:34:23.634Z,2021-06-21T07:34:23.203Z | subchannel | xxx READY -> IDLE
2021-06-21T07:34:23.634Z,2021-06-21T07:34:23.203Z | subchannel | xxx connection closed by GOAWAY with code 0
2021-06-21T07:34:21.634Z,2021-06-21T07:34:21.078Z | connectivity_state | dns:pubsub.googleapis.com:443 READY -> IDLE
2021-06-21T07:34:21.634Z,2021-06-21T07:34:21.078Z | resolving_load_balancer | dns:pubsub.googleapis.com:443 READY -> IDLE
2021-06-21T07:34:21.634Z,2021-06-21T07:34:21.078Z | pick_first | READY -> IDLE
2021-06-21T07:34:21.634Z,2021-06-21T07:34:21.078Z | subchannel_refcount | xxx refcount 2 -> 1
2021-06-21T07:34:21.634Z,2021-06-21T07:34:21.077Z | subchannel | xxx READY -> IDLE
2021-06-21T07:34:21.634Z,2021-06-21T07:34:21.077Z | subchannel | xxx connection closed by GOAWAY with code 0
2021-06-21T07:30:39.962Z,2021-06-21T07:30:39.742Z | subchannel_refcount | 2404:6800:4008:c03::5f:443 refcount 1 -> 0
2021-06-21T07:30:31.947Z,2021-06-21T07:30:31.333Z | call_stream | [2] pushing to reader message of length 53
2021-06-21T07:30:31.947Z,2021-06-21T07:30:31.333Z | call_stream | [2] filterReceivedMessage of length 58
2021-06-21T07:30:31.947Z,2021-06-21T07:30:31.333Z | call_stream | [2] parsed message of length 58
2021-06-21T07:30:31.947Z,2021-06-21T07:30:31.332Z | call_stream | [2] receive HTTP/2 data frame of length 58
2021-06-21T07:30:31.947Z,server-timing: gfet4t7; dur=7
2021-06-21T07:30:31.947Z,"date: Mon, 21 Jun 2021 07:30:31 GMT"
2021-06-21T07:30:31.947Z,"accept-encoding: identity,gzip"
2021-06-21T07:30:31.947Z,"grpc-accept-encoding: identity,deflate,gzip"
2021-06-21T07:30:31.947Z,content-type: application/grpc
2021-06-21T07:30:31.947Z,content-disposition: attachment
2021-06-21T07:30:31.947Z,:status: 200
2021-06-21T07:30:31.947Z,2021-06-21T07:30:31.249Z | call_stream | [2] Received server headers:
2021-06-21T07:30:31.947Z,2021-06-21T07:30:31.240Z | call_stream | [2] calling end() on HTTP/2 stream
2021-06-21T07:30:31.947Z,2021-06-21T07:30:31.239Z | call_stream | [2] sending data chunk of length 399 (deferred)
2021-06-21T07:30:31.947Z,2021-06-21T07:30:31.239Z | subchannel_refcount | xxx callRefcount 0 -> 1
2021-06-21T07:30:31.947Z,2021-06-21T07:30:31.239Z | call_stream | [2] attachHttp2Stream from subchannel xxx
2021-06-21T07:30:31.947Z,te: trailers
2021-06-21T07:30:31.947Z,:path: /google.spanner.v1.Spanner/ExecuteStreamingSql
2021-06-21T07:30:31.947Z,:method: POST
2021-06-21T07:30:31.947Z,content-type: application/grpc
2021-06-21T07:30:31.947Z,user-agent: grpc-node-js/1.3.2
2021-06-21T07:30:31.947Z,:authority: spanner.googleapis.com:443
2021-06-21T07:30:31.947Z,accept-encoding: identity
2021-06-21T07:30:31.947Z,"grpc-accept-encoding: identity,deflate,gzip"
2021-06-21T07:30:31.947Z,grpc-timeout: 17999799m
2021-06-21T07:30:31.947Z,authorization: xxx
2021-06-21T07:30:31.947Z,x-goog-request-params: xxx
2021-06-21T07:30:31.947Z,google-cloud-resource-prefix: xxx
2021-06-21T07:30:31.947Z,x-goog-api-client: gax/2.14.1 gapic/5.8.0 gl-node/14.15.4 grpc/ gccl/5.8.0
```
</details>

In failure case, server sent GOAWAY after sent 200 response for `ExecuteStreamingSql` request (partitioned update).

**Comment by murgatroid99** [comment]
Those logs are hard to read because they're in reverse order, and because the subchannel addresses are necessary to track what events are associated with each other, so redacting the addresses makes the logs ambiguous. But what I'm reading in the failure log is that the server never sent the status at the end of the request. If it had, there would have been a log line that said "Received server trailers". I don't think that the GOAWAY 4 whole minutes later is the issue.

5/11 does correspond to the release of grpc-js 1.3.1, but I don't currently see anything in that release that would affect this functionality.

**Comment by murgatroid99** [comment]
Various changes have been made to the connection management code in grpc-js since the last activity on this issue. I think it is likely that this problem no longer occurs. If you find this issue because you experienced a problem that appears the same with a recent version of the library, please file a new issue.

**Comment by akornatskyy** [comment]
after remote service restart the following continues in the loop and never connects:

```log
D 2023-11-16T15:06:47.180Z | v1.9.10 7026 | resolving_load_balancer | dns:192.168.56.1:5000 CONNECTING -> IDLE
D 2023-11-16T15:06:47.180Z | v1.9.10 7026 | channel | (1) dns:192.168.56.1:5000 callRefTimer.unref | configSelectionQueue.length=0 pickQueue.length=0
D 2023-11-16T15:06:47.180Z | v1.9.10 7026 | load_balancing_call | [50] Pick called
D 2023-11-16T15:06:47.180Z | v1.9.10 7026 | load_balancing_call | [50] Pick result: QUEUE subchannel: null status: undefined undefined
D 2023-11-16T15:06:47.180Z | v1.9.10 7026 | channel | (1) dns:192.168.56.1:5000 callRefTimer.ref | configSelectionQueue.length=0 pickQueue.length=1
D 2023-11-16T15:06:47.180Z | v1.9.10 7026 | connectivity_state | (1) dns:192.168.56.1:5000 CONNECTING -> IDLE
D 2023-11-16T15:06:47.180Z | v1.9.10 7026 | resolving_load_balancer | dns:192.168.56.1:5000 IDLE -> CONNECTING
D 2023-11-16T15:06:47.180Z | v1.9.10 7026 | channel | (1) dns:192.168.56.1:5000 callRefTimer.unref | configSelectionQueue.length=0 pickQueue.length=0
D 2023-11-16T15:06:47.180Z | v1.9.10 7026 | load_balancing_call | [50] Pick called
D 2023-11-16T15:06:47.180Z | v1.9.10 7026 | load_balancing_call | [50] Pick result: QUEUE subchannel: null status: undefined undefined
D 2023-11-16T15:06:47.180Z | v1.9.10 7026 | channel | (1) dns:192.168.56.1:5000 callRefTimer.ref | configSelectionQueue.length=0 pickQueue.length=1
D 2023-11-16T15:06:47.180Z | v1.9.10 7026 | connectivity_state | (1) dns:192.168.56.1:5000 IDLE -> CONNECTING
```

**Comment by murgatroid99** [comment]
@akornatskyy Do you have the trace logs from when the client started doing that?

**Comment by akornatskyy** [comment]
here is the case with 100% CPU (that continues in the loop until killed):

```log
D 2023-11-16T18:14:38.039Z | v1.9.10 1577 | subchannel_refcount | (10) /run/containerd/containerd.sock refcount 3 -> 4
D 2023-11-16T18:14:38.039Z | v1.9.10 1577 | subchannel_refcount | (10) /run/containerd/containerd.sock refcount 4 -> 3
D 2023-11-16T18:14:38.039Z | v1.9.10 1577 | pick_first | TRANSIENT_FAILURE -> TRANSIENT_FAILURE
D 2023-11-16T18:14:38.039Z | v1.9.10 1577 | resolving_load_balancer | unix:///run/containerd/containerd.sock TRANSIENT_FAILURE -> TRANSIENT_FAILURE
D 2023-11-16T18:14:38.040Z | v1.9.10 1577 | connectivity_state | (7) unix:///run/containerd/containerd.sock TRANSIENT_FAILURE -> TRANSIENT_FAILURE
D 2023-11-16T18:14:38.040Z | v1.9.10 1577 | subchannel_refcount | (10) /run/containerd/containerd.sock refcount 3 -> 4
D 2023-11-16T18:14:38.040Z | v1.9.10 1577 | subchannel_refcount | (10) /run/containerd/containerd.sock refcount 4 -> 3
D 2023-11-16T18:14:38.040Z | v1.9.10 1577 | pick_first | TRANSIENT_FAILURE -> TRANSIENT_FAILURE
```

**Comment by murgatroid99** [comment]
@akornatskyy Those should both be fixed in version 1.9.11

**Comment by akornatskyy** [comment]
@murgatroid99 : yes, both issues fixed with 1.9.11. many thanks.
